metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizer.java

```diff
@@ -138,6 +139,15 @@ public List<AuthorizationResult> authorize(
         return results;
     }
 
+    @Override
+    public AuthorizationResult authorizeByResourceType(
```

Comment:
If I understand correctly, this implementation was added to take advantage of the new binary search approach in the ACL array. IOW, an optimization over the default `authorizeByResourceType` impl?

Reply:
mainly I added this because the benchmarks looked really bad without it 😬